py/embedded-server/deephaven_server/server.py

```diff
@@ -7,10 +7,69 @@
 
 from .start_jvm import start_jvm
 
-# This is explicitly not a JObjectWrapper, as that would require importing deephaven and jpy
+# These classes are explicitly not JObjectWrapper, as that would require importing deephaven and jpy
 # before the JVM was running.
 
 
+class ServerConfig:
+    """
+    Represents the configuration of a Deephaven server.
+    """
+
+    def __init__(self, j_server_config):
+        self.j_server_config = j_server_config
+
+    @property
+    def j_object(self):
+        return self.j_server_config
+
+    @property
+    def target_url_or_default(self) -> str:
+        """
+        Returns the target URL to bring up the Web UI.
+
+        Returns:
+            The target URL to bring up the Web UI.
+        """
+        return self.j_server_config.targetUrlOrDefault()
+
+
+class AuthenticationHandler:
+    """
+    Represents an authentication handler for a Deephaven server.
+    """
+    def __init__(self, j_authentication_handler):
+        self.j_authentication_handler = j_authentication_handler
+
+    @property
+    def j_object(self):
+        return self.j_authentication_handler
+
+    @property
+    def auth_type(self) -> str:
+        """
+        Get the authentication type for this handler.
+
+        Returns:
+            The authentication type for this handler.
+        """
+        return self.j_authentication_handler.getAuthType()
+
+    def urls(self, target_url: str) -> List[str]:
```

Comment:
@devinrsmith Also this gives the URL to the IDE - just wondering, if we want URLs to the iframe endpoints as well, that could be useful for deephaven-ipywidgets (and others).

Reply:
This API doesn't assume much about `target_url`; in the case of anonymous, it just echos it back, and in the case of PSK, just appends `?psk=<key>`. It doesn't know anything about IDE nor iframe, etc.

By default, `target_url = "http://localhost:10000"`. As part of CLI, you _could_ append `/ide/` to be a bit more explicit, or invoke the logic multiple times to get iframe or other links.

It would be nice to have a more proper API around web apps installed as opposed to doing:

```java
for (String appRoot : List.of("/ide/", "/iframe/table/", "/iframe/chart/", "/iframe/widget/")) {
```

Until then though, I think you'll need to do something similar in CLI if you want to.